samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -170,6 +196,8 @@ public void onNewJobModelAvailable(final String version) {
     debounceTimer.scheduleAfterDebounceTime(ScheduleAfterDebounceTime.JOB_MODEL_VERSION_CHANGE, 0, () ->
       {
         LOG.info("pid=" + processorId + "new JobModel available");
+        metrics.newJobModel.inc();
```

Comment:
Can you please explain the significance of the "new-jobmodel-generated" metric? Does it indicate the number of times jobmodel was generated or the number of times this processor was notified about a jobmodel change?

If it is the former, I think this metric should be incremented only by the leader. If it is the latter, perhaps rename the metric as "new-jobmodel-available" ?